BLSQ/iaso
# POLIO-662 BWE country specific teams

Allow having team per country (e.g. GPEI or MoH) in a Budget Workflow. 

Certain step of the Workflow  are to be done by specific team from a Country, which can only do theses steps for campaigns from this country.
The way we solve this is to have the Workflow definition specify a Team of Teams, instead of a simple Team of Users (like we did till now), and then on the Polio country configuration page, we can specify which teams belong to which country.

In the workflow processing, if we encounter a Team of Team (either for permission or sending mail), we check all the descendants teams to filter the one belonging to the country (of the campaign).


## Self proof reading checklist

## Changes

- Add hierarchy() method on Team queryset, that work like on OrgUnit.
- Add teams on CountryUserGroup
- Show it in the UI
- filter on theses group for checking if a user can do the permission and when sending mail. (see description above)

## How to test

1. You can load a pre made Workflow config using Fixture.
    See wiki for instruction https://wiki.bluesquare.org/en/Dev-team/product-management/iaso/polio_budget_workflow
2. Create a Team of teams (or modify team 6) and then a team of user as a child of it. Put your user in it.
3. Take a campaign for test
4. Go to http://localhost:8081/dashboard/polio/config (you will need the polio admin perm) And change the configuration on  the Country of your Campaign to add your new team.
5. If you created a new team, reference it in the workflow.
6. Do campaign steps and check if everything work correctly.



## Print screen / video

![image](https://user-images.githubusercontent.com/82500/200353092-66810a5b-a9d9-4ca0-9a20-b5587b0c4f51.png)

## Notes
This PR is based on the POLIO-658 PR (Budget Widget) that's awaiting merging. So will need to be merged before.
